src/process/mod.rs

```diff
@@ -1528,6 +1531,7 @@ pub fn all_processes_with_root(root: impl AsRef<Path>) -> ProcResult<ProcessesIt
 /// for more information.
 #[derive(Debug)]
 pub struct ProcessesIter {
+    root: String,
```

Comment:
This should probably be `PathBuf` or `OsString`, because a path on Linux can be any bytes, whereas a String is valid UTF-8.

Reply:
Indeed, `PathBuf` is better, I changed it, thank you !

If you are ok with the last commit I will squash it into the "Fix process iterator with a custom root" one.

Reply:
I squashed it !

CI is broken because of the version in Cargo.toml, I don't think I can change that.

Reply:
There are multiple issues in CI atm, but not related to your changes

- rust 1.48 has an issue with the manifest format
- stable and nightly have an issue with sys::kernel::random::tests::test_poolsize. It seems related to a recent change in the kernel, see https://unix.stackexchange.com/questions/704737/kernel-5-10-119-caused-the-values-of-proc-sys-kernel-random-entropy-avail-and-p

However, I don't know if the change of failure to anyhow should be handled in this PR (I'm not the repo owner, though)

Reply:
As "failure" is used only in test and is deprecated, I thought it worst nothing to add it here.

If it is a problem I will of course remove it from the PR. The main issue is the real important thing to fix.

Thank you for your review !